source/math/RandomGenerator.ooc

```diff
@@ -0,0 +1,140 @@
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with this program. If not, see <http://www.gnu.org/licenses/>.
+import math
+use ooc-collections
+import os/Time
+
+RandomGenerator: abstract class {
+	_state: Int
+	init: func {
+		this _state = Time microtime()
+	}
+	init: func ~withSeed (=_state)
+	_next: func -> Float {
+		this _state = 214013 * this _state + 2531011
+		((((this _state >> 16) & 0x7fff) % 10000) as Float) / 10000.0f
+	}
+}
+
+FloatRandomGenerator: abstract class extends RandomGenerator {
+	init: func {
+		super()
+	}
+	init: func ~withSeed (seed: Int) {
+		super(seed)
+	}
+	next: abstract func -> Float
+	next: func ~withCount (count: Int) -> Float[] {
+		result := Float[count] new()
+		for (i in 0 .. count)
+			result[i] = this next()
+		result
+	}
+}
+
+FloatUniformRandomGenerator: class extends FloatRandomGenerator {
+	_min := 0.0f
+	_max := 1.0f
+	init: func {
+		super()
+	}
+	init: func ~withSeed (seed: Int) {
+		super(seed)
+	}
+	init: func ~withParameters (=_min, =_max, seed := Time microtime()) {
+		super(seed)
+	}
+	next: func -> Float {
+		Float linearInterpolation(this _min, this _max, this _next())
+	}
+}
+
+FloatGaussianRandomGenerator: class extends FloatRandomGenerator {
+	_mu := 0.0f
+	_sigma := 1.0f
+	init: func {
+		super()
+	}
+	init: func ~withSeed (seed: Int) {
+		super(seed)
+	}
+	init: func ~withParameters (=_mu, =_sigma, seed := Time microtime()) {
+		super(seed)
+	}
+	next: func -> Float {
+		first := this _next()
+		if (first == 0.0f)
+			first = Float minimumValue
+		second := this _next()
+		value := sqrt(-2.0f * log(first)) * cos(2.0f * PI * second)
+		value * this _sigma + this _mu
+	}
+}
+
+IntRandomGenerator: abstract class extends RandomGenerator {
+	init: func {
+		super()
+	}
+	init: func ~withSeed (seed: Int) {
+		super(seed)
+	}
+	next: abstract func -> Int
+	next: func ~withCount (count: Int) -> Int[] {
+		result := Int[count] new()
+		for (i in 0 .. count)
+			result[i] = this next()
+		result
+	}
+}
+
+IntUniformRandomGenerator: class extends IntRandomGenerator {
+	_min := 0
+	_max := 100
+	init: func {
+		super()
+	}
+	init: func ~withSeed (seed: Int) {
+		super(seed)
+	}
+	init: func ~withParameters (=_min, =_max, seed := Time microtime()) {
+		super(seed)
+	}
+	next: func -> Int {
+		this _next() * (this _max - this _min) + this _min
+	}
+}
+
+IntGaussianRandomGenerator: class extends IntRandomGenerator {
+	_mu := 0.0f
+	_sigma := 10.0f
+	init: func {
+		super()
+	}
+	init: func ~withSeed (seed: Int) {
+		super(seed)
+	}
+	init: func ~withParameters (=_mu, =_sigma, seed := Time microtime()) {
+		super(seed)
+	}
+	next: func -> Int {
+		first := this _next()
+		if (first == 0.0f)
+			first = Float minimumValue
+		second := this _next()
+		value := sqrt(-2.0f * log(first)) * cos(2.0f * PI * second)
+		(value * this _sigma + this _mu) as Int
+	}
```

Comment:
This function is identical to the one in `FloatGaussianRandomGenerator`, but casts the result to an Int. @simonmika, I know you were against having function names like `nextGaussian`, but what if we 

- create a function `_nextGaussian` in the `RandomGenerator` base class, 
- move the code in `FloatGaussianRandomGenerator next` there, and
- let both `IntGaussianRandomGenerator` and `FloatGaussianRandomGenerator` return the value from `RandomGenerator _nextGaussian` instead? The API will still look the same to anyone outside this module, but there'll be less duplicate code.